File: src/ddd/simple/service/permission/OperatorService.java
package ddd.simple.service.permission;


import java.util.ArrayList;
import java.util.List;
import java.util.Map;

import ddd.base.helper.RedirectPage;
import ddd.base.persistence.EntitySet;
import ddd.simple.entity.member.Member;
import ddd.simple.entity.organization.Organization;
import ddd.simple.entity.permission.Operator;
import ddd.simple.entity.permission.Role;
import ddd.simple.service.base.BaseServiceInterface;


public interface OperatorService extends BaseServiceInterface{

	public Operator saveOperator(Operator operator)throws Exception;

	public int deleteOperator(Long operatorId)throws Exception;

	public Operator updateOperator(Operator operator)throws Exception;

	public Operator findOperatorById(Long operatorId)throws Exception;

	public EntitySet<Operator> findAllOperators()throws Exception;
	
	public Operator findOperatorByCode(String operatorCode)throws Exception;
	
	public ArrayList<Organization> searchOrganization(String operatorCode)throws Exception;

	public Map<String, Object> checkLoginUser(String operatorCode,String operatorPassword,Organization organization) throws Exception;
	
	public boolean checkOperator(String operatorCode,String operatorPassword)throws Exception;
	
	public Integer distributionRole(Long operatorId,EntitySet<Role> roles,Long organizationId) throws Exception;

	public Integer copyOperator(List<Long> sourceEIds, List<Long> targetEIds) throws Exception;

	public Integer changePasswd(Long operatorId, String newPasw, String oldPasw) throws Exception;

	public Operator memberToOperator(Member member);
	
	public RedirectPage SSO(String userKey);
	
	public Map<String, Object> checkSSO(Long operatorId);
	
	public Operator findOperatorByEmployeeId(Long employeeId);
}

File: WebContent/ddd/simple/view/permission/js/operator/OperatorService.js
angular.module("operatorModule",[])

.factory('OperatorService',['$resource', function($resource)
{
	var operatorService = $resource('../Operator/:action', {});

	operatorService.saveOperator = function(operator,sucesscb, errorcb)
	{
		operatorService.save({action : "saveOperator"},{operator:operator}, sucesscb, errorcb);
	};
	
	operatorService.deleteOperator=function(operatorId,sucesscb,errorcb)
	{
		operatorService.save({action:"deleteOperator"},{operatorId:operatorId},sucesscb,errorcb);
	};
	
	operatorService.updateOperator=function(operator,sucesscb,errorcb)
	{
		operatorService.save({action:"updateOperator"},{operator:operator},sucesscb,errorcb);
	};

	operatorService.findOperatorById = function(operatorId,sucesscb, errorcb)
	{
		operatorService.save({action : "findOperatorById"},{operatorId:operatorId}, sucesscb, errorcb);
	};
	
	operatorService.findOperatorByCode  = function(operatorCode,sucesscb, errorcb)
	{
		operatorService.save({action : "findOperatorByCode"},{operatorCode:operatorCode}, sucesscb, errorcb);
	};
	
	operatorService.findAllOperators = function(sucesscb, errorcb)
	{
		operatorService.save({action : "findAllOperators"}, sucesscb, errorcb);
	};
	
	operatorService.distributionRole = function(operatorId,roles,organizationId)
	{
		return operatorService.save({action : "distributionRole"},{operatorId:operatorId,organizationId:organizationId,roles:roles}).$promise;
	};
	
	operatorService.copyOperator = function(sourceEIds,targetEIds,sucesscb, errorcb)
	{
		operatorService.save({action : "copyOperator"},{sourceEIds:sourceEIds,targetEIds:targetEIds}, sucesscb, errorcb);
	};
	
	
	operatorService.changePasswd = function(operatorId,newPasw,oldPasw,sucessCB,errorCB)
	{
		operatorService.save({action : "changePasswd"},{operatorId:operatorId,newPasw:newPasw,oldPasw:oldPasw},sucessCB,errorCB);
	}
	
	return operatorService;
}]);